src/main/java/com/minecolonies/core/util/RandomCollection.java

```diff
@@ -0,0 +1,96 @@
+package com.minecolonies.core.util;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.*;
+import java.util.Map.Entry;
+
+/**
+ * A collection that is able to generate a random item selection, based on weighted items.
+ * <a href="https://stackoverflow.com/questions/6409652/random-weighted-selection-in-java">Source</a>
+ *
+ * @param <K> the collection key.
+ * @param <E> the collection type.
+ */
+public class RandomCollection<K, E>
```

Comment:
I am pretty sure this collection already exists in Minecraft, known as a weighted list.

Could you check?

Reply:
Found it, WeightedRandomList, implemented it.